{
  try {
    foo();
  }
 catch (  FileNotFoundException e) {
    throw e;
  }
catch (  IOException e) {
    throw INSTANCE;
  }
}
